### PR TITLE
Loopback: add typing to Role.registerResolver function

### PR DESCRIPTION
**Pull request by hiddestokvis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Find an example usage of registerResolver here](https://loopback.io/doc/en/lb3/Defining-and-using-roles.html) [API docs here](http://apidocs.loopback.io/loopback/#role-registerresolver)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by enko** [comment]
@hiddestokvis please also add the promise based resolver:

```typescript
static registerResolver(role: string, resolver: (role: string, ctx: Context) => Promise<boolean>): void;
```

File: types/loopback/index.d.ts [hunk 1]
```diff
@@ -2653,7 +2653,7 @@ declare namespace l {
              * if a principal is in the specified role.
              * Should provide a callback or return a promise.
              */
-            static registerResolver(role: string, resolver: () => void): void;
+            static registerResolver(role: string, resolver: (role: string, ctx: Context, callback?: (err: Error, resolved: boolean) => void) => void): void;
```
Comment (enko) [inline types/loopback/index.d.ts]: Please add promised based resolver.

**Review by enko — CHANGES_REQUESTED** [review]

**Review by kattsushi — APPROVED** [review]
LGTM

**Comment by hiddestokvis** [comment]
@enko good catch, I made the changes

**Comment by Sequoia** [comment]
@ryancavanaugh is it possible for me to be removed from the loopback typings? I no longer work on that project. Thank you!

**Comment by andy-ms** [comment]
@Sequoia Create a pull request removing yourself from the header.

**Pull request merged**